### PR TITLE
Add CRTP safety to ModelMetrics and convenience method to

### PR DESCRIPTION
**Pull request by akleeman**
This minor change puts some safety checks and extra utility methods to the metrics.

### CODE REVIEW

File: include/albatross/src/evaluation/prediction_metrics.hpp [hunk 1]
```diff
@@ -71,13 +71,29 @@ struct RootMeanSquareError : public PredictionMetric<Eigen::VectorXd> {
...
+
+  double output = 0.;
+  double mean = x.mean();
+  const auto n_elements = static_cast<double>(x.size());
+
+  for (Eigen::Index i = 0; i < x.size(); ++i) {
+    output += pow(x[i] - mean, 2.) / (n_elements - 1);
+  }
+
+  return std::sqrt(output);
```
Comment (seth-swiftnav) [inline include/albatross/src/evaluation/prediction_metrics.hpp]: What's the reason behind this change from the prior implementation with `dot`?

Comment (akleeman) [reply include/albatross/src/evaluation/prediction_metrics.hpp]: I didn't actually run into this issue ... but it's possible that summing all the squared values before averaging could lead to a loss in numerical precision.  By dividing iteratively you avoid this.
